Another CMAKE issue on latest version.

I am an early adopter of this software and it has always worked exceptionally well.  However, I now cannot install this latest version.  Details follow:
Hardware:
    Raspberry Pi4 running the latest version of Raspbian
    WaveShare revision 2.1 board
    <note:  this is the same hardware and OS I have previously used with success>

Install process and errors as follows:
jon@raptor:~ $ rm -rf indi-wmh-focuser
jon@raptor:~ $ ver
bash: ver: command not found
jon@raptor:~ $ version
bash: version: command not found
jon@raptor:~ $ sudo apt install build-essential cmake libgpiod-dev
build-essential is already the newest version (12.12).
cmake is already the newest version (3.31.6-2).
libgpiod-dev is already the newest version (2.2.1-2).
Summary:
  Upgrading: 0, Installing: 0, Removing: 0, Not Upgrading: 0
jon@raptor:~ $ sudo apt install libindi-dev
libindi-dev is already the newest version (1.9.9+dfsg-3+b5).
Summary:
  Upgrading: 0, Installing: 0, Removing: 0, Not Upgrading: 0
jon@raptor:~ $ git clone https://github.com/k-ross/indi-wmh-focuser.git
cd indi-wmh-focuser
mkdir build
cd build
Cloning into 'indi-wmh-focuser'...
remote: Enumerating objects: 195, done.
remote: Counting objects: 100% (195/195), done.
remote: Compressing objects: 100% (147/147), done.
remote: Total 195 (delta 114), reused 112 (delta 47), pack-reused 0 (from 0)
Receiving objects: 100% (195/195), 98.38 KiB | 2.34 MiB/s, done.
Resolving deltas: 100% (114/114), done.
jon@raptor:~/indi-wmh-focuser/build $ cmake -DCMAKE_INSTALL_PREFIX=/usr ..
make
sudo make install
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found INDI: /usr/lib/aarch64-linux-gnu/libindidriver.so;/usr/lib/aarch64-linux-gnu/libindiAlignmentDriver.so (found version "1.9.9")
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1")
-- Checking for one of the modules 'libgpiod'
-- Checking for one of the modules 'libgpiodcxx'
-- Configuring done (3.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/jon/indi-wmh-focuser/build
[ 33%] Building CXX object CMakeFiles/indi_wmh_focuser.dir/wmh_focuser.cpp.o
In file included from /usr/include/gpiod.hpp:27,
                 from /home/jon/indi-wmh-focuser/gpiomotor.h:3,
                 from /home/jon/indi-wmh-focuser/wmh_focuser.cpp:65:
/usr/include/gpiodcxx/chip.hpp:45:36: error: ‘filesystem’ in namespace ‘std’ does not name a type; did you mean ‘system’?
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                    ^~~~~~~~~~
      |                                    system
/usr/include/gpiodcxx/chip.hpp:45:52: error: expected unqualified-id before ‘&’ token
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                                    ^
/usr/include/gpiodcxx/chip.hpp:45:52: error: expected ‘)’ before ‘&’ token
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                      ~                             ^
      |                                                    )
/usr/include/gpiodcxx/chip.hpp:45:52: error: constructors may not be ref-qualified
/usr/include/gpiodcxx/chip.hpp:45:52: error: expected ‘;’ at end of member declaration
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                                    ^
      |                                                     ;
/usr/include/gpiodcxx/chip.hpp:45:54: error: ‘path’ does not name a type
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                                      ^~~~
/usr/include/gpiodcxx/chip.hpp:84:16: error: ‘filesystem’ in namespace ‘std’ does not name a type; did you mean ‘system’?
   84 |         ::std::filesystem::path path() const;
      |                ^~~~~~~~~~
      |                system
In file included from /usr/include/gpiodcxx/line-request.hpp:20,
                 from /usr/include/gpiod.hpp:36:
/usr/include/gpiodcxx/misc.hpp:25:38: error: ‘filesystem’ in namespace ‘std’ does not name a type; did you mean ‘system’?
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                                      ^~~~~~~~~~
      |                                      system
/usr/include/gpiodcxx/misc.hpp:25:54: error: expected unqualified-id before ‘&’ token
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                                                      ^
/usr/include/gpiodcxx/misc.hpp:25:54: error: expected ‘)’ before ‘&’ token
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                        ~                             ^
      |                                                      )
/usr/include/gpiodcxx/misc.hpp:25:56: error: expected initializer before ‘path’
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                                                        ^~~~
/home/jon/indi-wmh-focuser/gpiomotor.h:13:12: error: ‘line’ in namespace ‘gpiod’ does not name a type
   13 |     gpiod::line _enableLine;
      |            ^~~~
In file included from /usr/include/gpiodcxx/chip.hpp:21:
/usr/include/gpiodcxx/line.hpp:24:11: note: ‘gpiod::line’ declared here
   24 | namespace line {
      |           ^~~~
/home/jon/indi-wmh-focuser/gpiomotor.h:14:12: error: ‘line’ in namespace ‘gpiod’ does not name a type
   14 |     gpiod::line _directionLine;
      |            ^~~~
/usr/include/gpiodcxx/line.hpp:24:11: note: ‘gpiod::line’ declared here
   24 | namespace line {
      |           ^~~~
/home/jon/indi-wmh-focuser/gpiomotor.h:15:12: error: ‘line’ in namespace ‘gpiod’ does not name a type
   15 |     gpiod::line _stepLine;
      |            ^~~~
/usr/include/gpiodcxx/line.hpp:24:11: note: ‘gpiod::line’ declared here
   24 | namespace line {
      |           ^~~~
make[2]: *** [CMakeFiles/indi_wmh_focuser.dir/build.make:79: CMakeFiles/indi_wmh_focuser.dir/wmh_focuser.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/indi_wmh_focuser.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
[ 33%] Building CXX object CMakeFiles/indi_wmh_focuser.dir/wmh_focuser.cpp.o
In file included from /usr/include/gpiod.hpp:27,
                 from /home/jon/indi-wmh-focuser/gpiomotor.h:3,
                 from /home/jon/indi-wmh-focuser/wmh_focuser.cpp:65:
/usr/include/gpiodcxx/chip.hpp:45:36: error: ‘filesystem’ in namespace ‘std’ does not name a type; did you mean ‘system’?
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                    ^~~~~~~~~~
      |                                    system
/usr/include/gpiodcxx/chip.hpp:45:52: error: expected unqualified-id before ‘&’ token
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                                    ^
/usr/include/gpiodcxx/chip.hpp:45:52: error: expected ‘)’ before ‘&’ token
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                      ~                             ^
      |                                                    )
/usr/include/gpiodcxx/chip.hpp:45:52: error: constructors may not be ref-qualified
/usr/include/gpiodcxx/chip.hpp:45:52: error: expected ‘;’ at end of member declaration
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                                    ^
      |                                                     ;
/usr/include/gpiodcxx/chip.hpp:45:54: error: ‘path’ does not name a type
   45 |         explicit chip(const ::std::filesystem::path& path);
      |                                                      ^~~~
/usr/include/gpiodcxx/chip.hpp:84:16: error: ‘filesystem’ in namespace ‘std’ does not name a type; did you mean ‘system’?
   84 |         ::std::filesystem::path path() const;
      |                ^~~~~~~~~~
      |                system
In file included from /usr/include/gpiodcxx/line-request.hpp:20,
                 from /usr/include/gpiod.hpp:36:
/usr/include/gpiodcxx/misc.hpp:25:38: error: ‘filesystem’ in namespace ‘std’ does not name a type; did you mean ‘system’?
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                                      ^~~~~~~~~~
      |                                      system
/usr/include/gpiodcxx/misc.hpp:25:54: error: expected unqualified-id before ‘&’ token
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                                                      ^
/usr/include/gpiodcxx/misc.hpp:25:54: error: expected ‘)’ before ‘&’ token
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                        ~                             ^
      |                                                      )
/usr/include/gpiodcxx/misc.hpp:25:56: error: expected initializer before ‘path’
   25 | bool is_gpiochip_device(const ::std::filesystem::path& path);
      |                                                        ^~~~
/home/jon/indi-wmh-focuser/gpiomotor.h:13:12: error: ‘line’ in namespace ‘gpiod’ does not name a type
   13 |     gpiod::line _enableLine;
      |            ^~~~
In file included from /usr/include/gpiodcxx/chip.hpp:21:
/usr/include/gpiodcxx/line.hpp:24:11: note: ‘gpiod::line’ declared here
   24 | namespace line {
      |           ^~~~
/home/jon/indi-wmh-focuser/gpiomotor.h:14:12: error: ‘line’ in namespace ‘gpiod’ does not name a type
   14 |     gpiod::line _directionLine;
      |            ^~~~
/usr/include/gpiodcxx/line.hpp:24:11: note: ‘gpiod::line’ declared here
   24 | namespace line {
      |           ^~~~
/home/jon/indi-wmh-focuser/gpiomotor.h:15:12: error: ‘line’ in namespace ‘gpiod’ does not name a type
   15 |     gpiod::line _stepLine;
      |            ^~~~
/usr/include/gpiodcxx/line.hpp:24:11: note: ‘gpiod::line’ declared here
   24 | namespace line {
      |           ^~~~
make[2]: *** [CMakeFiles/indi_wmh_focuser.dir/build.make:79: CMakeFiles/indi_wmh_focuser.dir/wmh_focuser.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/indi_wmh_focuser.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
jon@raptor:~/indi-wmh-focuser/build $

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,8 +36,9 @@ if(ROCKPI)
     )
     set(EXTRA_LIBS ${MRAA_LIBRARIES})
 else()
-    pkg_search_module(GPIOD REQUIRED libgpiod)
-    pkg_search_module(GPIODCXX REQUIRED libgpiodcxx)
+    pkg_check_modules(GPIOD REQUIRED libgpiod>=2.0)
+    pkg_check_modules(GPIODCXX REQUIRED libgpiodcxx>=2.0)
+    include_directories(${GPIODCXX_INCLUDE_DIRS})
     set(indi_wmh_focuser_SRCS
         wmh_focuser.cpp
         gpiomotor.cpp

--- a/README.md
+++ b/README.md
@@ -11,6 +11,12 @@ First, let's install some build tools, if you haven't already.
 sudo apt install build-essential cmake libgpiod-dev
 ```
 
+**Note:** This project requires **libgpiod v2.2+** (available in Debian Trixie and later). The user running `indiserver` needs access to `/dev/gpiochip*` devices. On most distributions, add your user to the `gpio` group:
+```
+sudo usermod -aG gpio $USER
+```
+Then log out and back in for the group change to take effect.
+
 If you built INDI from source, you can skip this step. Otherwise, you must install the INDI development libraries.
 ```
 sudo apt install libindi-dev

--- a/gpiomotor.cpp
+++ b/gpiomotor.cpp
@@ -1,76 +1,111 @@
 #include "gpiomotor.h"
+#include <iostream>
 
 using namespace std;
 
-GpioMotor::GpioMotor(const string& consumer, 
-        const string& enableChip, int enablePin, 
-        const string& directionChip, int directionPin, 
+filesystem::path GpioMotor::chipNameToPath(const string& name)
+{
+    if (!name.empty() && name[0] == '/')
+        return filesystem::path(name);
+    return filesystem::path("/dev") / name;
+}
+
+gpiod::line_request GpioMotor::requestOutputLine(
+    const string& chipName, int pin, const string& consumer)
+{
+    gpiod::line_settings settings;
+    settings.set_direction(gpiod::line::direction::OUTPUT);
+
+    return gpiod::chip(chipNameToPath(chipName))
+        .prepare_request()
+        .set_consumer(consumer)
+        .add_line_settings(gpiod::line::offset(pin), settings)
+        .do_request();
+}
+
+GpioMotor::GpioMotor(const string& consumer,
+        const string& enableChip, int enablePin,
+        const string& directionChip, int directionPin,
         const string& stepChip, int stepPin)
+    : _enablePin(enablePin), _directionPin(directionPin), _stepPin(stepPin)
 {
-    _enableChip.open(enableChip);
-    _directionChip.open(directionChip);
-    _stepChip.open(stepChip);
+    try {
+        _enableRequest.emplace(requestOutputLine(enableChip, enablePin, consumer));
+    } catch (const exception& e) {
+        cerr << "GpioMotor: failed to request enable line (chip="
+             << enableChip << " pin=" << enablePin << "): " << e.what() << endl;
+    }
 
-    if (_enableChip)    _enableLine     = _enableChip.get_line(enablePin);
-    if (_directionChip) _directionLine  = _directionChip.get_line(directionPin);
-    if (_stepChip)      _stepLine       = _stepChip.get_line(stepPin);
+    try {
+        _directionRequest.emplace(requestOutputLine(directionChip, directionPin, consumer));
+    } catch (const exception& e) {
+        cerr << "GpioMotor: failed to request direction line (chip="
+             << directionChip << " pin=" << directionPin << "): " << e.what() << endl;
+    }
 
-    gpiod::line_request request
-    {
-        .consumer = consumer,
-        .request_type = gpiod::line_request::DIRECTION_OUTPUT
-    };
-
-    if (_enableLine)    _enableLine.request(request);
-    if (_directionLine) _directionLine.request(request);
-    if (_stepLine)      _stepLine.request(request);
+    try {
+        _stepRequest.emplace(requestOutputLine(stepChip, stepPin, consumer));
+    } catch (const exception& e) {
+        cerr << "GpioMotor: failed to request step line (chip="
+             << stepChip << " pin=" << stepPin << "): " << e.what() << endl;
+    }
 }
-	
-GpioMotor::~GpioMotor()
-{
-}
-	
+
+GpioMotor::~GpioMotor() = default;
+
 void GpioMotor::Enable(Direction dir)
 {
-    if (_revision == BoardRevision::Rev21)
-        _enableLine.set_value (1);
-    else
-        _enableLine.set_value (0);
-        
-    if (dir == Direction::Forward)
-        _directionLine.set_value (0);
-    else
-        _directionLine.set_value (1);
+    if (_enableRequest) {
+        if (_revision == BoardRevision::Rev21)
+            _enableRequest->set_value(_enablePin, gpiod::line::value::ACTIVE);
+        else
+            _enableRequest->set_value(_enablePin, gpiod::line::value::INACTIVE);
+    }
+
+    if (_directionRequest) {
+        if (dir == Direction::Forward)
+            _directionRequest->set_value(_directionPin, gpiod::line::value::INACTIVE);
+        else
+            _directionRequest->set_value(_directionPin, gpiod::line::value::ACTIVE);
+    }
 }
 
 void GpioMotor::Disable()
 {
-    if (_revision == BoardRevision::Rev21)
-        _enableLine.set_value (0);
-    else
-        _enableLine.set_value (1);
+    if (_enableRequest) {
+        if (_revision == BoardRevision::Rev21)
+            _enableRequest->set_value(_enablePin, gpiod::line::value::INACTIVE);
+        else
+            _enableRequest->set_value(_enablePin, gpiod::line::value::ACTIVE);
+    }
 }
 
 void GpioMotor::SingleStep(int stepDelayMicroseconds)
 {
-    _stepLine.set_value (1);
-    _Delay (stepDelayMicroseconds);
-    _stepLine.set_value (0);
-    _Delay (stepDelayMicroseconds);
+    if (_stepRequest) {
+        _stepRequest->set_value(_stepPin, gpiod::line::value::ACTIVE);
+        _Delay(stepDelayMicroseconds);
+        _stepRequest->set_value(_stepPin, gpiod::line::value::INACTIVE);
+        _Delay(stepDelayMicroseconds);
+    }
 }
 
-//figure out what the chip name should be for Raspberry Pi
-//for Pi5 it should be gpiochip4; otherwise it should be gpiochip0
-//the chip.label() should return something like 'pinctrl_xxx', so we'll use that to detect
+// Figure out what the chip name should be for Raspberry Pi.
+// For Pi5 it should be gpiochip4; otherwise it should be gpiochip0.
+// The chip label should return something like 'pinctrl_xxx', so we use that to detect.
 string GpioMotor::getPiChip()
 {
-    static std::string piChip;
+    static string piChip;
     if (piChip.empty()) {
         piChip = "gpiochip0";
-        for (auto& it: ::gpiod::make_chip_iter()) {
-            string substr = "pinctrl";
-            if (strncmp(it.label().c_str(), substr.c_str(), substr.size())==0) {
-                piChip = it.name();
+        for (const auto& entry : filesystem::directory_iterator("/dev/")) {
+            if (gpiod::is_gpiochip_device(entry.path())) {
+                gpiod::chip chip(entry.path());
+                auto info = chip.get_info();
+                string label = info.label();
+                if (label.rfind("pinctrl", 0) == 0) {
+                    piChip = info.name();
+                }
             }
         }
     }

--- a/gpiomotor.h
+++ b/gpiomotor.h
@@ -1,23 +1,24 @@
 #pragma once
 
 #include <gpiod.hpp>
-#include <memory>
-#include <cstring>
+#include <optional>
+#include <filesystem>
+#include <string>
 #include "motor.h"
 
 class GpioMotor : public Motor
 {
-    gpiod::chip _enableChip;
-    gpiod::chip _directionChip;
-    gpiod::chip _stepChip;
-    gpiod::line _enableLine;
-    gpiod::line _directionLine;
-    gpiod::line _stepLine;
-    
+    std::optional<gpiod::line_request> _enableRequest;
+    std::optional<gpiod::line_request> _directionRequest;
+    std::optional<gpiod::line_request> _stepRequest;
+    gpiod::line::offset _enablePin;
+    gpiod::line::offset _directionPin;
+    gpiod::line::offset _stepPin;
+
 public:
-    GpioMotor(const std::string& consumer, 
-        const std::string& enableChip, int enablePin, 
-        const std::string& directionChip, int directionPin, 
+    GpioMotor(const std::string& consumer,
+        const std::string& enableChip, int enablePin,
+        const std::string& directionChip, int directionPin,
         const std::string& stepChip, int stepPin);
 
     ~GpioMotor() override;
@@ -25,6 +26,11 @@ public:
     void Enable(Direction dir) override;
     void Disable() override;
     void SingleStep(int stepDelayMicroseconds) override;
-    
+
     static std::string getPiChip();
+
+private:
+    static std::filesystem::path chipNameToPath(const std::string& name);
+    static gpiod::line_request requestOutputLine(
+        const std::string& chipName, int pin, const std::string& consumer);
 };

